src/system_utils.jl

```diff
@@ -21,3 +93,99 @@ function (sm::SystemMap{DT})(u0, p) where {DT}
                       p = convert(typeof(sm.prob.p), p))
     solve(prob, sm.args...; sm.kwargs...)
 end
+
+function ExpectationProblem(sm::SystemMap, g, h, d; nout = 1)
+    ExpectationProblem(sm, g, h, d,
+                       ArrayPartition(deepcopy(sm.prob.u0), deepcopy(sm.prob.p)),
+                       nout)
+end
+
+"""
+```julia
+ProcessNoiseSystemMap(prob, args...; kwargs...)
+```
+
+Representation of a system solution map for a given `prob::DEProblem`. `args` and `kwargs`
+are forwarded to the equation solver.
+"""
+struct ProcessNoiseSystemMap{DT <: DiffEqBase.DEProblem, A, K} <: AbstractSystemMap
+    prob::DT
+    n::Int
+    args::A
+    kwargs::K
+end
+ProcessNoiseSystemMap(prob, n, args...; kwargs...) = ProcessNoiseSystemMap(prob, n, args, kwargs)
+
+function (sm::ProcessNoiseSystemMap{DT})(Z, p) where {DT}
+    W(t) = sqrt(2) * sum(Z[k] * sin((k - 0.5) * pi * t) / ((k - 0.5) * pi) for k in 1:length(Z))
+    prob::DT = remake(sm.prob, p = convert(typeof(sm.prob.p), p),
+                               noise = NoiseFunction{false}(prob.tspan[1],W))
+    solve(prob, sm.args...; sm.kwargs...)
+end
+
+function ExpectationProblem(sm::ProcessNoiseSystemMap, g, h; nout = 1)
+    d = GenericDistribution((Normal() for i in 1:sm.n)...)
+    ExpectationProblem(sm, g, h, d, deepcopy(sm.prob.p), nout)
+end
```

Comment:
This assumes the noise process os `WienerProcess`. Is the thought to eventually dispatch on noise type in `SDEProblem`? Would it make any sense to have a `KLExpansions.jl` to return the distribution and noise process to make it easy to extend to other noise processes?

Reply:
I think it would be easiest to define those dispatches on `ProcessNoiseSystemMap` because most of the different forms are best represented just by a `NoiseFunction`, and so it's just pre-defined `NoiseFunction` forms. Maybe some can be pushed back to DiffEqNoiseProcess.jl